src/App.tsx

```diff
@@ -17,18 +80,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onChangeStatus={handleChangeStatus}
+                query={query}
+                onChangeQuery={handleChangeQuery}
+                onReset={onReset}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              <TodoList
+                todos={filteredTodos}
+                showSelectedTodo={handelTodo}
+                selectedTodo={selectedTodo}
+              />
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          user={user}
+          todo={selectedTodo}
+          loading={loading}
```

Comment:
Recommendation: The TodoModal component should have a personal loading state. When you click on the Todo button, the loadings duplicate.

Reply:
> Recommendation: The TodoModal component should have a personal loading state. When you click on the Todo button, the loadings duplicate.

Good point